src/tests/conduit/t_conduit_node_convert.cpp

```diff
@@ -55,84 +55,72 @@
+    CONDUIT_NATIVE_UNSIGNED_CHAR_ID, CONDUIT_NATIVE_UNSIGNED_SHORT_ID, CONDUIT_NATIVE_UNSIGNED_INT_ID, CONDUIT_NATIVE_UNSIGNED_LONG_ID,
+    CONDUIT_NATIVE_FLOAT_ID, CONDUIT_NATIVE_DOUBLE_ID
+};
+
+const NodeConvertFun CONVERT_TO_FUNS[20] = {
+    convert_to_int8_array, convert_to_int16_array, convert_to_int32_array, convert_to_int64_array,
+    convert_to_uint8_array, convert_to_uint16_array, convert_to_uint32_array, convert_to_uint64_array,
+    convert_to_float32_array, convert_to_float64_array,
+    convert_to_char_array, convert_to_short_array, convert_to_int_array, convert_to_long_array,
+    convert_to_unsigned_char_array, convert_to_unsigned_short_array, convert_to_unsigned_int_array, convert_to_unsigned_long_array,
+    convert_to_float_array, convert_to_double_array
+};
+
 //-----------------------------------------------------------------------------
 TEST(conduit_node_convert, to_arrays)
 {
-    uint8 data_vals[] = {1,2,3,4,5,6,7,8};
-    
-    Node n;
-    n.set(data_vals,8);
-
-    n.schema().print();
-    
-    Node nconv;
-    
-    // signed bw-style
-    n.to_int8_array(nconv);
-    nconv.print();
-
-    n.to_int16_array(nconv);
-    nconv.print();
-
-    n.to_int32_array(nconv);
-    nconv.print();
-    
-    n.to_int64_array(nconv);
-    nconv.print();
-
-    // unsigned bw-style
-    n.to_uint8_array(nconv);
-    nconv.print();
-
-    n.to_uint16_array(nconv);
-    nconv.print();
-
-    n.to_uint32_array(nconv);
-    nconv.print();
-    
-    n.to_uint64_array(nconv);
-    nconv.print();
-    
-    // float bw-style
-    n.to_float32_array(nconv);
-    nconv.print();
-    
-    n.to_float64_array(nconv);
-    nconv.print();
-
-    // signed native c
-    n.to_char_array(nconv);
-    nconv.print();
-
-    n.to_short_array(nconv);
-    nconv.print();
-
-    n.to_int_array(nconv);
-    nconv.print();
-    
-    n.to_long_array(nconv);
-    nconv.print();
-
-    // unsigned native c
-    n.to_unsigned_char_array(nconv);
-    nconv.print();
-
-    n.to_unsigned_short_array(nconv);
-    nconv.print();
-
-    n.to_unsigned_int_array(nconv);
-    nconv.print();
-    
-    n.to_unsigned_long_array(nconv);
-    nconv.print();
-
-    // float native c
-    n.to_float_array(nconv);
-    nconv.print();
-    
-    n.to_double_array(nconv);
-    nconv.print();
-
-
+    uint8 data_vals[3] = {1,2,3};
+    Node data_node;
+    data_node.set(data_vals,3);
+
+    for(index_t ti = 0; ti < 20; ti++)
+    {
+        int node_tid = CONVERT_TYPES[ti];
+        NodeConvertFun convert_fun = CONVERT_TO_FUNS[ti];
+
+        Node type_node;
+        convert_fun(data_node, type_node);
+
+        EXPECT_EQ(type_node.dtype().id(), node_tid);
+        EXPECT_EQ(type_node.dtype().number_of_elements(),
+            data_node.dtype().number_of_elements());
+
+        Node temp_node;
+        convert_to_uint8_array(type_node, temp_node);
+        EXPECT_FALSE(memcmp(temp_node.data_ptr(), data_node.data_ptr(),
```

Comment:
I am confused b/c I feel like there should be one case where its basically uint8 to uint8, and this should pass?